pandas/tests/window/test_rolling.py

```diff
@@ -1950,3 +1950,24 @@ def test_numeric_only_corr_cov_series(kernel, use_arg, numeric_only, dtype):
         op2 = getattr(rolling2, kernel)
         expected = op2(*arg2, numeric_only=numeric_only)
         tm.assert_series_equal(result, expected)
+
+
+def test_rolling_rolling_sum_window_microseconds_confilict_timestamp():
+    # GH#55106
+    df_time = DataFrame(
+        {"B": [0, 1, 2, 4, 5, 6]},
+        index=[
+            Timestamp("20130101 09:00:00"),
+            Timestamp("20130101 09:00:02"),
+            Timestamp("20130101 09:00:03"),
+            Timestamp("20130101 09:00:06"),
+            Timestamp("20130101 09:00:07"),
+            Timestamp("20130101 09:00:08"),
+        ],
+    )
```

Comment:
I know this is the example from the issue report, but now we are writing the test, we could actually slightly simplify the creation here, to simplify the test code:

```suggestion
    df = DataFrame({"A": range(5)}, index=date_range("2013-01-01", freq="1s", periods=5))
```